### ISSUE TITLE
Decouple implicit build → test relationship

### ISSUE THREAD

**Issue by TomasTomecek** [packit/packit-service#1807]
As we are heading in the direction of having multiple jobs that can consume artifacts from other (hello vm-image-build), we should decouple the implicit relationship of `copr_build` and `tests` jobs.

The expectation here is that:
- tests job needs a copr_build job if RPMs are consumed there: https://github.com/packit/packit-service/issues/1775
- `/packit copr_build` doesn't trigger tests
- the relationship needs to be explicitly referenced: https://github.com/packit/packit-service/issues/1773

**Comment by lachmanfrantisek** [comment]
+1 for the decoupling, it will make the code really simpler!

> * `/packit copr_build` doesn't trigger tests

Hmm,  I am wondering how to do this. And if this is really needed. (Test submission is done when we receive a build result.)

**Comment by TomasTomecek** [comment]
> +1 for the decoupling, it will make the code really simpler!
> 
> > * `/packit copr_build` doesn't trigger tests
> 
> Hmm, I am wondering how to do this. And if this is really needed. (Test submission is done when we receive a build result.)

You are right! The point, it should be configurable :) The "jobs referencing" should solve it.

**Comment by mfocko** [comment]
> +1 for the decoupling, it will make the code really simpler!
> 
> > * `/packit copr_build` doesn't trigger tests
> 
> Hmm, I am wondering how to do this. And if this is really needed. (Test submission is done when we receive a build result.)

Maybe we're getting to the point where _job trigger_ is not actually a job trigger and we want _job trigger_ and _job source_ :smile: /s

ps: :dagger: sounds good

**Comment by lachmanfrantisek** [comment]
> Maybe we're getting to the point where job trigger is not actually a job trigger and we want job trigger and job source smile /s

Exactly!
Do we want to solve that here? Or do this separately? (Since it's valid to all the downstream jobs as well.

**Comment by nforro** [comment]
Still relevant but low prio.
